KZL-557 - Add collection controller csharp (#191)
kuzzleio/documentation-V2: src/sdk-reference/csharp/1/collection/validate-specifications/index.md

---
layout: sdk.html.hbs
title: validateSpecifications

> description: Validate specifications format

Reviewer: ```suggestion
description: Validates specifications format
```

---

# validateSpecifications

> The validateSpecifications method checks if a validation specification is well formatted. It does not store nor modify the existing specification.

Reviewer 2: ```suggestion
Checks if a validation specification is well formatted. It does not store nor modify the existing specification.
```

When the validation specification is not formatted correctly, a detailed error message is returned to help you to debug.

## Signature

```csharp
public validation_response validateSpecifications(
    string index, 
    string collection, 
    string specifications);

public validation_response validateSpecifications(
    string index, 
    string collection, 
    string specifications, 
    query_options options);

```

## Arguments

| Arguments    | Type    | Description |
|--------------|---------|-------------|
| `index` | <pre>string</pre> | Index name    | 
| `collection` | <pre>string</pre> | Collection name    |
| `specifications` | <pre>string<pre> | JSON string representating the specifications to validate |
| `options` | <pre>Kuzzleio::QueryOptions\*</pre> | Query options    | 

### specifications

A JSON string representing the specifications the specifications.

The JSON must follow the [Specification Structure]({{ site_base_path }}guide/1/datavalidation):

> ```json

Reviewer: ```suggestion
```js
```

{
  "strict": "true",
  "fields": {
    "licence": {
      "mandatory": true,
      "type": "string"
    },
    // ... specification for each field
  }
}
```

### options

Additional query options

| Property     | Type<br/>(default)    | Description        |
| ---------- | ------- | --------------------------------- | 
| `queuable` | <pre>bool</pre><br/>(`true`) | If true, queues the request during downtime, until connected to Kuzzle again |

## Return

> A `kuzzleio::validation_response` object containing information about the specifications validity.

Reviewer: ```suggestion
A `Kuzzleio::validation_response` object containing information about the specifications validity.
```

> A `kuzzleio::validation_response` object which contain information about the specifications validity.

Reviewer 2: ```suggestion
A `kuzzleio::validation_response` object containing information about the specifications validity.
```
